R/derive_param_tte.R

```diff
@@ -142,10 +159,261 @@
+#'   censor_conditions = list(eos),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTSAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First Serious", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Using different censor values (`censor`)
+#' @info Within `event_source()`/`censor_source()` the values used to denote
+#'   an event/censor can be changed from the default of `0`/`1`.
+#'
+#' In this example an extra censor is used for new drug date with the value of `2`.
+#' @code
+#' newdrug <- censor_source(
+#'   dataset_name = "adsl",
+#'   date = NEWDRGDT,
+#'   censor = 2,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "NEW DRUG RECEIVED",
+#'     SRCDOM = "ADSL",
+#'     SRCVAR = "NEWDRGDT"
+#'   )
+#' )
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos, newdrug),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @info In this case the results are still the same as the latest censor
+#'   condition is always taken for those without an event, and for the second
+#'   subject this is still the end of study date.
+#'   So, if we wanted to instead censor here at the new drug date if subject
+#'   has one, then we would need to again use the `filter` argument, but this
+#'   time for a new end of study censor source object.
+#' @code
+#' eos_nonewdrug <- censor_source(
+#'   dataset_name = "adsl",
+#'   filter = is.na(NEWDRGDT),
+#'   date = EOSDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "END OF STUDY",
+#'     SRCDOM = "ADSL",
+#'     SRCVAR = "EOSDT"
+#'   )
+#' )
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos_nonewdrug, newdrug),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Overall survival time to event parameter
+#' @info In oncology trials, this is commonly derived as time from randomization
+#'   date to death. For those without event, they are censored at the last date
+#'   they are known to be alive.
+#'
+#' - The start date is set using `start_date` argument, now that we need to use
+#'   different to the default.
+#' - In this example, datetime was needed, which can be achieved by setting
+#'   `create_datetime` argument to `TRUE`.
+#' @code
 #' data("admiral_adsl")
 #'
 #' adsl <- admiral_adsl
```

Comment:
I would create a small ADSL dataset here. Then the input is visible to the reader and we don't need the `filter()` call below.

Reply:
sure